machiavelli-cards/src/main/java/pl/jellytech/machiavelli/cards/controllers/CardController.java

```diff
@@ -0,0 +1,122 @@
+package pl.jellytech.machiavelli.cards.controllers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.ObjectFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+import pl.jellytech.machiavelli.cards.dtos.BaseResponseWrapper;
+import pl.jellytech.machiavelli.cards.dtos.CardResponse;
+import pl.jellytech.machiavelli.cards.entities.Card;
+import pl.jellytech.machiavelli.cards.entities.CardType;
+import pl.jellytech.machiavelli.cards.services.CardService;
+import pl.jellytech.machiavelli.cards.utils.ControllerUtils;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+@RestController()
+@RequestMapping( "api/card")
+@Slf4j
+public class CardController {
+    private final CardService cardService;
+    private final ModelMapper modelMapper;
+    @Autowired
+    public CardController(CardService cardService, ModelMapper modelMapper){
+
+        this.cardService = cardService;
+        this.modelMapper = modelMapper;
+    }
+
+    @PostMapping(consumes ={MediaType.MULTIPART_FORM_DATA_VALUE} )
+    public ResponseEntity createOrUpdate(@RequestPart("cardType") String cardType,
+                                                              @RequestPart("name") String name,
+                                                              @RequestPart("description") String description,
+                                                              @RequestPart(value = "cardId",required = false) Optional<Long> cardId,
+                                                              @RequestPart("image") MultipartFile image){
+
+        try{
+            final byte[] imageBytes = image.getBytes();
+            Supplier<Card> func = () -> this.cardService.createOrUpdate(new Card(
+                    CardType.valueOf(cardType.toUpperCase()),
+                    name,
+                    imageBytes,
+                    description,
+                    cardId
+            ));
+            final Card card = ControllerUtils.FunctionLogMeasureWrapper(func,
+                    String.format("Saving Card entity with name %s",name),
```

Comment:
spacing - for formatting please look for a ready to use IntelliJ formatter and add it (it can be used as a plugin so it will automatically format on build)

Reply:
there is one which sync formatting from eclipse. I have used built-in but it's  (said pretty) not ok